uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -743,15 +773,23 @@ define([
             listToReturn = listOfDayTypes.filter(function (day) {
               return day.name === 'non_working_day';
             });
-          } else if (this.calendar.isWeekend(moment(date))) {
-            listToReturn = listOfDayTypes.filter(function (day) {
-              return day.name === 'weekend';
-            });
           }
         } catch (e) {
           listToReturn = [];
         }
 
+        if (!listToReturn.length) {
+          try {
+            if (this.calendar.isWeekend(moment(date))) {
+              listToReturn = listOfDayTypes.filter(function (day) {
+                return day.name === 'weekend';
+              });
+            }
+          } catch (e) {
+            listToReturn = [];
+          }
+        }
+
```

Comment:
I'm not sure if i get exactly what is going on here, but wouldn't all of that be equivalent to something like
```js
/**
 * Gets list of day types if its found to be weekend or non working in calendar
 *
 * @param {Date} date to Checks
 * @param {Array} listOfDayTypes array of day types
 * @return {Array} non-empty if found else empty array
 */
function getDayTypesFromDate(date, listOfDayTypes) {
  var nameFilter = null;

  if (this.calendar.isNonWorkingDay(moment(date))) {
    nameFilter = 'non_working_day';
  } else if (this.calendar.isWeekend(moment(date))) {
    nameFilter = 'weekend';
  }

  return !nameFilter ? [] : listOfDayTypes.filter(function (day) {
    return day.name === nameFilter;
  });
}
```